### PR TITLE
leddy: Init at 0.1.0-unstable-2024-04-05

### PR DESCRIPTION
**Pull request by jmir1**
https://github.com/XanClic/leddy

A program for cotrolling the LEDs of Fnatic Streak and miniStreak keyboards.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/le/leddy/package.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  buildInputs = [
+    hidapi
+    udev
+  ];
+
+  meta = with lib; {
+    description = "LED controller for the Fnatic miniStreak and Fnatic Streak keyboards.";
+    homepage = "https://github.com/XanClic/leddy";
+    license = licenses.gpl3Plus;
+    maintainers = [ maintainers.jmir ];
```
Comment (FliegendeWurst) [inline pkgs/by-name/le/leddy/package.nix:31]:
```suggestion
  meta = {
    description = "LED controller for the Fnatic miniStreak and Fnatic Streak keyboards";
    homepage = "https://github.com/XanClic/leddy";
    license = lib.licenses.gpl3Plus;
    maintainers = [ lib.maintainers.jmir ];
```

File: pkgs/by-name/le/leddy/package.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    hash = "sha256-7t+E47odtayw26AnhtkxIWr0TxDwruEjP3Af3ajmVAA=";
+  };
+  cargoHash = "sha256-99qPo5Nsu8pLMT5EIpQ/+F0o/00Y5gQ9T0z5+gxUcww=";
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [
+    hidapi
+    udev
+  ];
+
```
Comment (FliegendeWurst) [inline pkgs/by-name/le/leddy/package.nix:26]:
```suggestion
  doCheck = false; # no tests
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/by-name/le/leddy/package.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [
+    hidapi
+    udev
+  ];
+  doCheck = false; # no tests
+
+  meta = {
+    description = "LED controller for the Fnatic miniStreak and Fnatic Streak keyboards.";
```
Comment (FliegendeWurst) [inline pkgs/by-name/le/leddy/package.nix:29]:
```suggestion
    description = "LED controller for the Fnatic miniStreak and Fnatic Streak keyboards";
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

**Review by FliegendeWurst — APPROVED** [review]

**Comment by jmir1** [comment]
Does this need another approval or can it be merged?

File: maintainers/maintainer-list.nix [hunk 4]
```diff
@@ -10420,6 +10420,12 @@
     githubId = 54179289;
     name = "Jason Miller";
   };
+  jmir = {
```
Comment (FliegendeWurst) [inline maintainers/maintainer-list.nix:10423]: Actually, this should be a separate commit. Then it will be merged soon

Comment (jmir1) [reply maintainers/maintainer-list.nix:10423]: Done :+1:

**Review by FliegendeWurst — APPROVED** [review]
